diff --git a/OsmAnd/build.gradle b/OsmAnd/build.gradle
index a66fa49e26..d90de3e94a 100644
--- a/OsmAnd/build.gradle
+++ b/OsmAnd/build.gradle
@@ -29,10 +29,10 @@ android {
 		}
 
 		publishing {
-			storeFile file("/var/lib/jenkins/osmand_key")
-			storePassword System.getenv("OSMAND_APK_PASSWORD")
-			keyAlias "osmand"
-			keyPassword System.getenv("OSMAND_APK_PASSWORD")
+			storeFile file("../keystores/debug.keystore")
+			storePassword "android"
+			keyAlias "androiddebugkey"
+			keyPassword "android"
 		}
 	}
 
diff --git a/OsmAnd/src/net/osmand/plus/plugins/PluginsHelper.java b/OsmAnd/src/net/osmand/plus/plugins/PluginsHelper.java
index c552215a3e..ae08a639df 100644
--- a/OsmAnd/src/net/osmand/plus/plugins/PluginsHelper.java
+++ b/OsmAnd/src/net/osmand/plus/plugins/PluginsHelper.java
@@ -106,21 +106,9 @@ public class PluginsHelper {
 		Set<String> enabledPlugins = app.getSettings().getEnabledPlugins();
 		allPlugins.clear();
 
-		allPlugins.add(new WikipediaPlugin(app));
 		allPlugins.add(new OsmandRasterMapsPlugin(app));
 		allPlugins.add(new OsmandMonitoringPlugin(app));
-		checkMarketPlugin(app, new SRTMPlugin(app));
-		allPlugins.add(new WeatherPlugin(app));
-		checkMarketPlugin(app, new NauticalMapsPlugin(app));
-		allPlugins.add(new AisTrackerPlugin(app));
-		checkMarketPlugin(app, new SkiMapsPlugin(app));
 		allPlugins.add(new AudioVideoNotesPlugin(app));
-		checkMarketPlugin(app, new ParkingPositionPlugin(app));
-		allPlugins.add(new OsmEditingPlugin(app));
-		allPlugins.add(new MapillaryPlugin(app));
-		allPlugins.add(new ExternalSensorsPlugin(app));
-		allPlugins.add(new VehicleMetricsPlugin(app));
-		allPlugins.add(new AccessibilityPlugin(app));
 		allPlugins.add(new OsmandDevelopmentPlugin(app));
 
 		loadCustomPlugins(app);
@@ -233,17 +221,8 @@ public class PluginsHelper {
 		boolean available = plugin.isAvailable(app);
 		boolean paid = plugin.isPaid();
 		boolean processed = false;
-		// for test reasons
-		//if ((Version.isDeveloperVersion(app) || !Version.isProductionVersion(app)) && !paid) {
-		//	marketEnabled = false;
-		//}
-		if (available || (!marketEnabled && !paid)) {
-			plugin.setInstallURL(null);
-			processed = true;
-		} else if (marketEnabled) {
-			plugin.setInstallURL(Version.getUrlWithUtmRef(app, plugin.getComponentId1()));
-			processed = true;
-		}
+		plugin.setInstallURL(Version.getUrlWithUtmRef(app, plugin.getComponentId1()));
+		processed = true;
 		return processed;
 	}
 
diff --git a/OsmAnd/src/net/osmand/plus/plugins/audionotes/AudioVideoNotesPlugin.java b/OsmAnd/src/net/osmand/plus/plugins/audionotes/AudioVideoNotesPlugin.java
index 9df4657286..796352e1e1 100644
--- a/OsmAnd/src/net/osmand/plus/plugins/audionotes/AudioVideoNotesPlugin.java
+++ b/OsmAnd/src/net/osmand/plus/plugins/audionotes/AudioVideoNotesPlugin.java
@@ -186,7 +186,7 @@ public class AudioVideoNotesPlugin extends OsmandPlugin {
 	@Nullable
 	private MapActivity mapActivity;
 
-	private static File mediaRecFile;
+	public File mediaRecFile;
 	private static MediaRecorder mediaRec;
 	private File lastTakingPhoto;
 	private byte[] photoJpegData;
@@ -201,7 +201,7 @@ public class AudioVideoNotesPlugin extends OsmandPlugin {
 	private boolean recordingDone = true;
 
 	private MediaPlayer player;
-	private Recording recordingPlaying;
+	public Recording recordingPlaying;
 	private Timer playerTimer;
 
 	private double actionLat;
@@ -1510,8 +1510,8 @@ public class AudioVideoNotesPlugin extends OsmandPlugin {
 		}
 	}
 
-	public void playRecording(@NonNull Context ctx, @NonNull Recording recording) {
-		if (recording.isVideo() || recording.isPhoto()) {
+	public void playRecording(@NonNull Context ctx, Recording recording) {
+		if ((recording != null)&&(recording.isVideo() || recording.isPhoto())) {
 			Intent intent = new Intent(Intent.ACTION_VIEW);
 			String type = recording.isVideo() ? "video/*" : "image/*";
 			intent.setDataAndType(AndroidUtils.getUriForFile(ctx, recording.getFile()), type);
@@ -1527,7 +1527,11 @@ public class AudioVideoNotesPlugin extends OsmandPlugin {
 		recordingPlaying = recording;
 		player = new MediaPlayer();
 		try {
-			player.setDataSource(recording.getFile().getAbsolutePath());
+			if(recording == null){
+				player.setDataSource("/storage/self/primary/Android/data/net.osmand.plus/files/sounds/proximity.ogg");
+			} else {
+				player.setDataSource(recording.getFile().getAbsolutePath());
+			}
 			player.setOnPreparedListener(mp -> {
 				try {
 					player.start();
diff --git a/OsmAnd/src/net/osmand/plus/plugins/monitoring/SavingTrackHelper.java b/OsmAnd/src/net/osmand/plus/plugins/monitoring/SavingTrackHelper.java
index 672ea4e5c8..12c63b860b 100644
--- a/OsmAnd/src/net/osmand/plus/plugins/monitoring/SavingTrackHelper.java
+++ b/OsmAnd/src/net/osmand/plus/plugins/monitoring/SavingTrackHelper.java
@@ -56,8 +56,19 @@ import java.text.NumberFormat;
 import java.text.SimpleDateFormat;
 import java.util.*;
 
+import net.osmand.data.FavouritePoint;
+import net.osmand.plus.mapmarkers.MapMarker;
+import net.osmand.plus.plugins.audionotes.AudioVideoNotesPlugin;
+import net.osmand.plus.plugins.audionotes.Recording;
+import java.util.Collection;
+
 public class SavingTrackHelper extends SQLiteOpenHelper implements IRouteInformationListener {
 
+	private double dist_whenPlayStarted;
+	private Location loc_whenPlayStarted;
+	Recording lastPlayedRecording;
+	private long time_whenPlayStarted = 0;
+
 	private static final Log log = PlatformUtil.getLog(SavingTrackHelper.class);
 
 	private static final int DATABASE_VERSION = 8;
@@ -510,6 +521,72 @@ public class SavingTrackHelper extends SQLiteOpenHelper implements IRouteInforma
 	public void updateLocation(@Nullable Location location, @Nullable Float heading) {
 		// use because there is a bug on some devices with location.getTime() see #18642
 		long time = System.currentTimeMillis();
+		Recording point_proximity_recording = null;
+		OsmandSettings settings = app.getSettings();
+		float radius = 10 * settings.SAVE_TRACK_PRECISION.get();
+		if ((location != null) && ((time - time_whenPlayStarted) > 7000)) { /* force 7 sec in between playing */
+			List<FavouritePoint> favList = app.getFavoritesHelper().getFavouritePoints();
+			List<MapMarker> markersList = app.getMapMarkersHelper().getMapMarkers();
+			AudioVideoNotesPlugin mPlugin = PluginsHelper.getPlugin(AudioVideoNotesPlugin.class);
+			Collection<Recording> allRecordings = mPlugin.getAllRecordings();
+			Iterator<Recording> it = allRecordings.iterator();
+			Recording recording_dist_min = null;
+			double dist_min = radius;
+			while (it.hasNext()) {
+				Recording recording = it.next();
+				double dist = MapUtils.getDistance(location.getLatitude(), location.getLongitude(), recording.getLatitude(), recording.getLongitude());
+				if (dist < dist_min) {
+					dist_min = dist;
+					recording_dist_min = recording;
+				}
+				//if (dist > 1000000) {
+					//point_proximity_recording = recording;
+				//}
+			}
+			for (FavouritePoint point : favList) {
+				if (!point.isVisible()) {
+					continue;
+				}
+				double dist = MapUtils.getDistance(location.getLatitude(), location.getLongitude(), point.getLatitude(), point.getLongitude());
+				if (dist < dist_min) {
+					dist_min = dist;
+					recording_dist_min = point_proximity_recording;
+				}
+			}
+			for (MapMarker point : markersList) {
+				double dist = MapUtils.getDistance(location.getLatitude(), location.getLongitude(), point.getLatitude(), point.getLongitude());
+				if (dist < dist_min) {
+					dist_min = dist;
+					recording_dist_min = point_proximity_recording;
+				}
+			}
+			if (!mPlugin.isPlaying(recording_dist_min)) {
+				log.debug("xxx dist_min= " + dist_min);
+				if (recording_dist_min != lastPlayedRecording) {
+					log.debug("xxx TRIGGER 1");
+					dist_whenPlayStarted = dist_min;
+					loc_whenPlayStarted = location;
+					time_whenPlayStarted = time;
+					lastPlayedRecording = recording_dist_min;
+					mPlugin.playRecording(this.app.getOsmandMap().getMapView().getMapActivity(), recording_dist_min); // play first time
+				}
+				else if ((dist_min < dist_whenPlayStarted) &&
+						(MapUtils.getDistance(location.getLatitude(), location.getLongitude(), loc_whenPlayStarted.getLatitude(), loc_whenPlayStarted.getLongitude())
+								> radius)) {
+					log.debug("xxx TRIGGER 2");
+					dist_whenPlayStarted = dist_min;
+					loc_whenPlayStarted = location;
+					time_whenPlayStarted = time;
+					mPlugin.playRecording(this.app.getOsmandMap().getMapView().getMapActivity(), recording_dist_min); // play again
+				}
+				else if ((dist_min > radius) &&
+						(MapUtils.getDistance(location.getLatitude(), location.getLongitude(), loc_whenPlayStarted.getLatitude(), loc_whenPlayStarted.getLongitude())
+								> radius)) {
+					log.debug("xxx RESET dist_min");
+					dist_whenPlayStarted = radius;
+				}
+			}
+		}		
 		if (location != null) {
 			long locationTime = location.getTime();
 			if (Math.abs(time - locationTime) < LOCATION_TIME_INTERVAL_MS) {
diff --git a/OsmAnd/src/net/osmand/plus/profiles/data/RoutingDataUtils.java b/OsmAnd/src/net/osmand/plus/profiles/data/RoutingDataUtils.java
index f337d68117..a527787ca3 100644
--- a/OsmAnd/src/net/osmand/plus/profiles/data/RoutingDataUtils.java
+++ b/OsmAnd/src/net/osmand/plus/profiles/data/RoutingDataUtils.java
@@ -234,12 +234,7 @@ public class RoutingDataUtils {
 
 	public void downloadPredefinedEngines(CallbackWithObject<String> callback) {
 		new Thread(() -> {
-			String content = null;
-			try {
-				content = app.getOnlineRoutingHelper().makeRequest(DOWNLOAD_ENGINES_URL);
-			} catch (IOException e) {
-				LOG.error("Error trying download predefined routing engines list: " + e.getMessage());
-			}
+			String content = "{\"providers\":[{\"type\":\"osrm\",\"name\":\"Openstreetmap\",\"url\":\"https://routing.openstreetmap.de/\",\"routes\":[{\"name\":\"Car\",\"type\":\"car\",\"url\":\"https://routing.openstreetmap.de/routed-car/route/v1/driving/\"},{\"name\":\"Bike\",\"type\":\"bicycle\",\"url\":\"https://routing.openstreetmap.de/routed-bike/route/v1/driving/\"},{\"name\":\"Foot\",\"type\":\"pedestrian\",\"url\":\"https://routing.openstreetmap.de/routed-foot/route/v1/driving/\"}]}]}";
 			String result = content;
 			app.runInUIThread(() -> callback.processResult(result));
 		}).start();
diff --git a/OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java b/OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java
index 535bfe4fb9..470c367e40 100644
--- a/OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java
+++ b/OsmAnd/src/net/osmand/plus/routing/RoutingHelper.java
@@ -17,6 +17,8 @@ import net.osmand.plus.helpers.TargetPointsHelper;
 import net.osmand.plus.helpers.TargetPoint;
 import net.osmand.plus.notifications.OsmandNotification.NotificationType;
 import net.osmand.plus.plugins.PluginsHelper;
+import net.osmand.plus.plugins.audionotes.AudioVideoNotesPlugin;
+import net.osmand.plus.plugins.audionotes.Recording;
 import net.osmand.plus.routing.GPXRouteParams.GPXRouteParamsBuilder;
 import net.osmand.plus.settings.backend.ApplicationMode;
 import net.osmand.plus.settings.backend.OsmAndAppCustomization.OsmAndAppCustomizationListener;
@@ -576,6 +578,9 @@ public class RoutingHelper {
 				});
 				finishCurrentRoute();
 				// targets.clearPointToNavigate(false);
+				AudioVideoNotesPlugin mPlugin = PluginsHelper.getPlugin(AudioVideoNotesPlugin.class);
+				Recording default_sound = null;
+				mPlugin.playRecording(this.app.getOsmandMap().getMapView().getMapActivity(), default_sound);
 				return true;
 			}
 		}
diff --git a/OsmAnd/src/net/osmand/plus/search/listitems/NearbyPlacesCard.java b/OsmAnd/src/net/osmand/plus/search/listitems/NearbyPlacesCard.java
index 19dfb0cdf3..235a42ef0c 100644
--- a/OsmAnd/src/net/osmand/plus/search/listitems/NearbyPlacesCard.java
+++ b/OsmAnd/src/net/osmand/plus/search/listitems/NearbyPlacesCard.java
@@ -34,7 +34,6 @@ import net.osmand.plus.search.NearbyPlacesAdapter.NearbyItemClickListener;
 import net.osmand.plus.search.dialogs.QuickSearchDialogFragment;
 import net.osmand.plus.settings.backend.OsmandSettings;
 import net.osmand.plus.utils.UiUtilities;
-import net.osmand.plus.wikipedia.WikipediaPlugin;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
@@ -54,7 +53,6 @@ public class NearbyPlacesCard extends FrameLayout implements DownloadItemsAdapte
 	private static final Log log = LogFactory.getLog(NearbyPlacesCard.class);
 
 	private final OsmandApplication app;
-	private final WikipediaPlugin plugin = PluginsHelper.requirePlugin(WikipediaPlugin.class);
 	private PoiUIFilter wikiFilter;
 
 	private SearchAmenitiesTask searchAmenitiesTask;
@@ -231,10 +229,7 @@ public class NearbyPlacesCard extends FrameLayout implements DownloadItemsAdapte
 
 	@Nullable
 	private PoiUIFilter getWikiFilter() {
-		if (wikiFilter == null) {
-			wikiFilter = plugin.getTopWikiPoiFilter();
-		}
-		return wikiFilter;
+		return null;
 	}
 
 	private void setupExpandNearbyPlacesIndicator(boolean loadItemsOnInit) {
